docs/backend-specialist/2.3-entity-api.md

```diff
@@ -0,0 +1,485 @@
+
+There are two entity variants:
+
+### Configuration Entities:
+- Used by the Configuration System.
+- Supports translations and can provide custom defaults for installations.
+- Exported as `.yml` files and stored in the config sync folder.
+- Can be imported into other environments.
+- Primarily has simple fields, making it difficult to integrate complex field widgets due to the use of a simple Form API element.
+
+### Content Entities:
+- Consist of configurable and base fields, can have revisions, and support translations.
+- Can have bundles, form modes, and display modes configured.
+- Field widgets and formatters can be set up, with support for Layout Builder.
+- Content moderation workflows can be applied.
+- Supports the Views module (though Views does not work with Config entities by default—there's a contrib module for that).
+- Stored in the database and not synced during regular deployments.
+- Site builders configure content entity types, while editors or content managers create the content.
+- Entities can also have subtypes, known as bundles.
+
+## When to Create a Content Entity Type vs. a Configuration Entity?
+
+**Configuration Entities:**
+- Stable and usually defined by developers.
+- They can be changed across environments using tools like Config Split, but by default, these entities are global among all environments.
+- They are stored in configuration files and should not be changed by customers, users, or editors.
+
+**Content Entities:**
+- Stored in the database and are fieldable, allowing site builders to add fields, manage form and view displays.
+- They differ between environments unless the database is synced.
+
+## Why Create Custom Entity Types Instead of Using Nodes for Everything?
+
+For simple websites, Drupal's default entities are usually sufficient. However, for complex systems, a well-architected solution may involve developing custom configuration and content entity types.
+
+The **Node** entity comes with many content-related fields and a lot of built-in functionality. For a clean architecture, it’s better to create a new content entity type with only the required fields. This allows developers to easily customize the entity, add Views handlers, and tweak the list builder and `save()` method without relying on hooks.
+
+The same applies to config entity types. Sometimes, simple configuration forms that save values to a single config file are not enough, especially when you need to create different configurations of the same structure/type.
+
+For example, if you need a system like **CompanyType**, it's not ideal to store it alongside content like **Countries** or **Tags**. System configurations should be static, and developers should refer to entities like **CompanyType** by machine name (available in custom entity types), not by term ID.
+
```

Comment:
More explanation?
In this example, `CompanyType` is a configuration entity type, which is static and should be referred to by its machine name. This is in contrast to content entities like `Countries` or `Tags`, which are more dynamic and typically referred to by term ID.